pandas/core/arrays/integer.py

```diff
@@ -196,7 +196,11 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             mask = mask.copy()
         return values, mask
 
-    values = np.array(values, copy=copy)
+    if hasattr(values, "dtype"):
```

Comment:
ok this needs a comment here, as this is typically exactly the reason we use ``np.asarray``

maybe bettetr to use
```
values = np.asarray(values, type=getattr(values, 'dtype', object), copy=copy)
```

Reply:
asarray is not working in a lot of cases. See the logs https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=26136&view=logs&j=bef1c175-2c1b-51ae-044a-2437c76fc339&t=770e7bb1-09f5-5ebf-b63b-578d2906aac9&l=159

Reply:
@jreback what should I do in this situation?

Reply:
Is `np.array` the only thing that works? If so I think OK to just revert to that for now